examples/paho-mqtt/Makefile

```
@@ -0,0 +1,58 @@
+APPLICATION = paho-mqtt-example
+
+# If no BOARD is found in the environment, use this default:
+BOARD ?= native
+
+# This has to be the absolute path to the RIOT base directory:
+RIOTBASE ?= $(CURDIR)/../..
+
+# Comment this out to disable code in RIOT that does safety checking
+# which is not needed in a production environment but helps in the
+# development process:
+DEVELHELP ?= 1
+
+# Change this to 0 show compiler invocation lines by default:
+QUIET ?= 1
+
+WIFI_SSID ?= "Your_WiFi_name"
+WIFI_PASS ?= "Your_secure_password"
+
+USEMODULE += shell
+USEMODULE += shell_commands
+USEMODULE += ps
+USEPKG += paho-mqtt
+
+# paho-mqtt depends on TCP support, choose the stack you want
+LWIP_IPV4 ?= 0
+
+ifneq (0, $(LWIP_IPV4))
+  USEMODULE += ipv4_addr
+  USEMODULE += lwip_arp
+  USEMODULE += lwip_ipv4
+  USEMODULE += lwip_dhcp_auto
+  USEMODULE += lwip_sock_udp
+  CFLAGS += -DETHARP_SUPPORT_STATIC_ENTRIES=1
+  LWIP_IPV6 ?= 0
+else
+  LWIP_IPV6 ?= 1
+endif
+
+ifneq (0, $(LWIP_IPV6))
+  USEMODULE += ipv6_addr
+  USEMODULE += lwip_ipv6_autoconfig
+endif
+
+USEMODULE += lwip_netdev
+USEMODULE += lwip lwip_sock_ip
+USEMODULE += lwip_tcp lwip_sock_tcp
+USEMODULE += lwip_sock_async
+
+USEMODULE += sock_async_event
+####
+
+include $(RIOTBASE)/Makefile.include
+
+ifneq (,$(filter arch_esp32,$(FEATURES_USED)))
+  CFLAGS += -DESP_WIFI_SSID=\"$(WIFI_SSID)\"
+  CFLAGS += -DESP_WIFI_PASS=\"$(WIFI_PASS)\"
+endif
```

With the following change, it can also be used with ESP8266 and `esp_wifi`.
```suggestion
CFLAGS += $(if $(filter esp_wifi,$(USEMODULE)),-DESP_WIFI_SSID=\"$(WIFI_SSID)\")
CFLAGS += $(if $(filter esp_wifi,$(USEMODULE)),-DESP_WIFI_PASS=\"$(WIFI_PASS)\")
```

---

Should we leave it like you propose, like that is gonna be available for any type of board or add esp8266 arch to the filter?

---

Shouldn't we add esp_wifi module that is being used only for esp32 now, in Makefile.board.dep ?

---

Good point, just add the same for `arch_esp8266` for the moment. From my point of view, it would be good to use the common ESP feature `arch_esp` and test for `arch_esp` but I don't know how fast it can be merged (PR #14375).

---

Yes, I saw it and it's a nice add. Let's wait a bit if your PR gets merged, and change the filter.